catalog/transformer/catalog-transformer-csv-common/src/main/java/ddf/catalog/transformer/csv/common/MetacardIterator.java

```diff
@@ -65,16 +84,82 @@ public Serializable next() {
 
     AttributeDescriptor attributeDescriptor = this.attributeDescriptorList.get(index);
     Attribute attribute = metacard.getAttribute(attributeDescriptor.getName());
+    AttributeFormat attributeFormat = attributeDescriptor.getType().getAttributeFormat();
     index++;
 
     if (attribute != null) {
       if (attributeDescriptor.isMultiValued()) {
-        return StringUtils.join(attribute.getValues(), MULTIVALUE_DELIMITER);
+        List<Serializable> convertedValues =
+            attribute.getValues().stream()
+                .map(value -> convertValue(attribute.getName(), value, attributeFormat))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        return StringUtils.join(convertedValues, MULTIVALUE_DELIMITER);
       } else {
-        return attribute.getValue();
+        Serializable value =
+            convertValue(attribute.getName(), attribute.getValue(), attributeFormat);
+        return (value == null) ? "" : value;
       }
+    } else if (isSourceId(attributeDescriptor) && isSourceIdSet()) {
+      return metacard.getSourceId();
+    } else if (isMetacardType(attributeDescriptor) && isMetacardTypeSet()) {
+      return metacard.getMetacardType().getName();
     }
 
     return "";
   }
+
+  private Serializable convertValue(
+      String name, Serializable value, AttributeType.AttributeFormat format) {
+    if (value == null) {
+      return null;
+    }
+
+    switch (format) {
+      case DATE:
+        if (!(value instanceof Date)) {
+          LOGGER.debug(
+              "Dropping attribute date value {} for {} because it isn't a Date object.",
+              value,
+              name);
+          return null;
+        }
+        Instant instant = ((Date) value).toInstant();
+        ZoneId zoneId = ZoneId.of("UTC");
+        ZonedDateTime zonedDateTime = instant.atZone(zoneId);
```

Comment:
✏️ I'm pretty sure the conversion to a `ZonedDateTime` is unnecessary. `Instant`'s `toString()` method will return an ISO-8601 string in UTC: https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#toString--